### PR TITLE
Add -launch flag to the build script

### PR DESCRIPTION
**Pull request by sharwell**
Example usage:

```text
.\build.cmd -release -deployExtensions -launch
```

Also made the `-deployExtensions` flag omit the build property unless false.

<details><summary>Ask Mode template not completed</summary>

### Customer scenario

What does the customer do to get into this situation, and why do we think this
is common enough to address for this release.  (Granted, sometimes this will be
obvious "Open project, VS crashes" but in general, I need to understand how
common a scenario is)

### Bugs this fixes

(either VSO or GitHub links)

### Workarounds, if any

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

### Risk

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

### Performance impact

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

### Root cause analysis

How did we miss it?  What tests are we adding to guard against it in the future?

### How was the bug found?

(E.g. customer reported it vs. ad hoc testing)

### Test documentation updated?

If this is a new non-compiler feature or a significant improvement to an existing feature, update https://github.com/dotnet/roslyn/wiki/Manual-Testing once you know which release it is targeting.

</details>

### CODE REVIEW

File: build/scripts/build.ps1 [hunk 1]
```diff
@@ -250,7 +251,7 @@ function Build-Artifacts() {
         Run-MSBuild "Compilers.sln" -useDotnetBuild
     }
     elseif ($build) {
-        Run-MSBuild "Roslyn.sln" "/p:DeployExtension=$deployExtensions"
+        Run-MSBuild "Roslyn.sln" $(if (-not $deployExtensions) {"/p:DeployExtension=false"})
```
Comment (jaredpar) [inline build/scripts/build.ps1]: Was not aware Powershell allowed for such complex interpolation. Did you find docs expressing the full limits of it?

File: build/scripts/build.ps1 [hunk 2]
```diff
@@ -29,6 +29,7 @@ param (
     [switch]$packAll = $false,
     [switch]$binaryLog = $false,
     [switch]$deployExtensions = $false,
+    [switch]$launch = $false,
```
Comment (jaredpar) [inline build/scripts/build.ps1]: Did you intend for `$launch` to be valid without `$deployExtensions`?

Comment (jaredpar) [inline build/scripts/build.ps1]: Consider adding an error to `Process-Arguments` when `$build` and `$launch` are specified but not `$deploy`.

Comment (sharwell) [reply build/scripts/build.ps1]: 💭 It seemed reasonable but could go either way.

File: build/scripts/build.ps1 (hunk 1 above)
Comment (sharwell) [reply build/scripts/build.ps1]: I found this: https://stackoverflow.com/a/34311158/138304

Then I tried simplifying it by omitting `else {""}` and found that it worked, so I left it.

File: build/scripts/build.ps1 (hunk 2 above)
Comment (sharwell) [reply build/scripts/build.ps1]: The alternative is to make `-build -launch` imply `-deployExtensions`. The advantage is the following can be used at that point:

```
.\build.cmd -release -launch
```

I can go either way. Let me know which change, if any, you prefer.

Comment (jaredpar) [reply build/scripts/build.ps1]: I actually liked the idea of `-launch` being completely independent. In the case of `-build -launch` it seems most likely the developer isn't doing what they think they're doing.

Comment (sharwell) [reply build/scripts/build.ps1]: ➡️ Validation is now added

**Pull request merged**